### ISSUE TITLE
More details on the Requests page

### ISSUE THREAD

**Issue by jboulter11** [getsentry/sentry#72628]
### Problem Statement

I want the Requests page to break down the requests by route, not just by domain.

Basically, it's cool that you can see each domain's high level stats but it's not that useful, really, until I can see it broken down by url.

I also wanna be able to setup alerts based off these things to catch outages or server/client regressions.

If you need screenshots here ping us in Slack, I can't post them publicly.

### Solution Brainstorm

I want to see that the /2/log_timing route is seeing 10% 500 responses 90% 200 responses.  I want to see exact error codes, so if a route is getting 422 vs 401s I wanna see that.

### Product Area

Performance

**Comment by jeengbe** [comment]
Hi, sharing my thoughts on this too 🙂 

Wouldn't this, especially the "setup alerts based off these things" be more suited for the service/project directly rather than a downstream service's dependency dashboard? How would that work with several services using one service, do alerts fire in both?

I imagine this feature more like an overview to understand which of the dependencies it is that's acting up, and to drill down further, that specific service's dashboard would open instead. 

Including individual routes on the requests dashboard would likely also clutter the view too much, but maybe you could have like a row you can expand, which then displays transactions from all grouped services.

**Comment by bcoe** [comment]
@jboulter11 thanks for taking the time to open an issue.

> Basically, it's cool that you can see each domain's high level stats but it's not that useful, really, until I can see it broken down by url.

> Including individual routes on the requests dashboard would likely also clutter the view too much, but maybe you could have like a row you can expand

We originally looked at doing routes rather than domain and bumped into some cardinality issues, e.g., it's not uncommon to have a URI with parameters that contain a lot of variability, e.g., resource identifiers, and you end up not finding good enough groupings to provide interesting aggregate information.

This is somewhat a limitation of the implementation, and I think it's a decision that can be revisited. I can imagine having a tab that breaks data down by **routes**, and a tab that breaks down data by **domain**.

> I also wanna be able to setup alerts based off these things to catch outages or server/client regressions.

@jeengbe I hear what you're saying, about _Requests Insights_, potentially being noisy for alerting...

However, I think it would be great to be able to setup alerts for any of the Insights landing pages in Sentry (_Requests_, _Queries_, _Caches_, _Queues_, _Web Vitals_).

This work is already tracked [here](https://github.com/getsentry/sentry/issues/72507).

**Comment by jboulter11** [comment]
> Wouldn't this, especially the "setup alerts based off these things" be more suited for the service/project directly rather than a downstream service's dependency dashboard? How would that work with several services using one service, do alerts fire in both?

Sure, but in reality those teams have no idea when they break my app and we need a way to tell them.  We're sometimes a small percentage of traffic, and almost nobody slices their traffic alerts to ensure the apps don't break.  Plus, it's possible requests get lost in routing layers and then any service alerting that might have caught something just doesn't know there's requests coming in at all.

> We originally looked at doing routes rather than domain and bumped into some cardinality issues.
I'd be nice if the same kind of thing that can strip the variables out of error messages for the grouping algorithm could be used to also strip those kinds of things out of the routes, then you'd be able to report and keep cardinality down.  Obviously I'd also want to strip the query params too.  

We have 500s instrumented from the iOS/Android SDK auto-instrumentation and we use this discover query to see when they're spiking, which route it is: https://dropbox.sentry.io/discover/results/?id=26942&project=4505546602381312&project=4505546600087552&statsPeriod=24h

However, this is too expensive when we do not need all the extra stuff provided in these 500s.  I'd rather it just be in traces and on the requests page as lower-cardinality info, and include 400s as well.  As it stands, I'm going to have to instrument this all myself (or try and piggy-back off Sentry SDK) to get it organized in a useful way we can look at and set alerts on.

Agreed that things like queries are also something I'd like to be able to set up alerts on.  "If this query performance gets worse, let me know" sort of thing.

**Comment by bcoe** [comment]
Blocked on https://github.com/getsentry/sentry/issues/81750 (which is in the works).

**Comment by mo-MikeJones** [comment]
+1 for this.  I think its quite common use case for an App to use 3rd party services, and then want to see the performance of requests to those services aggregated up.

Looking at the top two levels of the domain isn't ideal either.  In a corporate environment where you are using many services developed by many other teams, the top two levels of the domain are often the same, as the organisation is the same.

**Comment by bcoe** [comment]
> +1 for this. I think its quite common use case for an App to use 3rd party services

@mo-MikeJones, agreed. We are working on new querying capabilities for Sentry's span telemetry that will make answering questions like this possible.

**Comment by jboulter11** [comment]
Can we get an explanation as to why this has been closed?  Last we had heard this was in the works and we even saw a demo of some progress.

**Comment by bcoe** [comment]
> Can we get an explanation as to why this has been closed?

Our ticketing program was automatically closing older issues. We have turned this off.

For what it's worth, you should already be able to achieve something like this today using [Trace Explorer with Span Metrics](https://docs.sentry.io/product/explore/trace-explorer/), and a query something like this:

<img width="2364" height="860" alt="Image" src="https://github.com/user-attachments/assets/b2745276-fb24-4343-8816-b158eddd4703" />
